### DISCUSSION TITLE
Enable sending messages to unsubscribed clients

### DISCUSSION THREAD

**Discussion by IvDmNe** [foxglove/community#842] [General]
Suppose, we have running websocket server that has 2 types of channels it translates to Studio:
 - Image or PointCloud data that primarily used only in 3D/Image panels;
 - Localization info, System diagnostics or other "time-series" data that is displayed in Plot panel.

Now when we change active tab in Studio, a history of plot messages is wiped out, client stops advertising, new messages are not sent from server. But for channels of second type it would be useful to continue receive messages to have all data in reopened Plot panels.

I think channels could have a parameter something like ```bool send_to_unadvertised``` that is being checked along with client being advertised?

https://github.com/foxglove/ws-protocol/blob/c9170c4965ab75f3468c9fc2b2c1554c1142e41c/cpp/foxglove-websocket/include/foxglove/websocket/websocket_server.hpp#L988

Do you think it's worth to be added in ws-protocol?

**Comment by jtbandes** [comment]
Closing as duplicate of #846.

**Reply by defunctzombie** [reply]
Additional note here is a future workflow we want to support is DVR on live connections. That would help some of the above.
